velox/exec/HashProbe.cpp

```diff
@@ -355,17 +356,23 @@ void HashProbe::fillOutput(vector_size_t size) {
       output_);
 }
 
-RowVectorPtr HashProbe::getNonMatchingOutputForRightJoin() {
-  if (!lastRightJoinProbe_) {
-    return nullptr;
-  }
-
+RowVectorPtr HashProbe::getBuildSideOutput() {
   outputRows_.resize(outputBatchSize_);
-  auto numOut = table_->listNotProbedRows(
-      &rightJoinIterator_,
-      outputBatchSize_,
-      RowContainer::kUnlimited,
-      outputRows_.data());
+  int32_t numOut;
+  if (isRightSemiJoin(joinType_)) {
```

Comment:
I can't spot the difference between if and else branches. Would you clarify?

Reply:
if branch has `listProbedRows` and else branch has `listNotProbedRows`